src/nemos/basis/_basis.py

```diff
@@ -570,13 +756,13 @@ def _get_feature_slicing(
         return split_dict, start_slice
 
     @staticmethod
-    def _generate_unique_key(existing_dict: dict, key: str) -> str:
+    def _generate_unique_key(existing_dict: dict | List | Tuple, key: str) -> str:
         """Generate a unique key if there is a conflict."""
         extra = 1
-        new_key = f"{key}-{extra}"
+        new_key = f"{key}_{extra}"
```

Comment:
Didn't we decide that `-` was better than `_`? because `__` gets used by sklearn and `_` is used in our attributes